Mismatch between flag type and function argument for `recvmmsg`

<!--
Please provide the below information:

* a Minimum Working Example without any dependencies (except libc) that shows
  the issue and ideally reproduces in the Rust playground
* the target triple - libc supports many targets and many APIs
* instructions to reproduce, logs (e.g. build logs, links to Travis-CI logs,
  uploads to github gists, etc.).

In general, just please consider that the people who can help you are all very
busy, they will be helping you in their free time, there are a lot of people in
need of help so they need to prioritize to which issues they devote their free
time, etc. So try to give most of the information upfront, be concise, show
small self-contained examples (nobody has time to create a new cargo project,
set up dependencies,...), etc. Help us help you.

-->

The function `recvmmsg` for `musl` and `emscripten` has argument type `c_uint` for flags, but the flag types as defined in this package are `c_int`.

Problematic function definitions here:
https://github.com/rust-lang/libc/blob/7ecea9ac6d4e2093a6619d2fc0fd3b843ee21aa4/src/unix/linux_like/linux/musl/mod.rs#L706
https://github.com/rust-lang/libc/blob/6eee580b51455131b10b780087025379a591ceb6/src/unix/linux_like/emscripten/mod.rs#L1873

I encounted this when compiling a dependency that used `libc`:

```
#18 580.4    Compiling solana-streamer v1.10.40
#18 580.8 error[E0308]: mismatched types
#18 580.8   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/solana-streamer-1.10.40/src/recvmmsg.rs:99:70
#18 580.8    |
#18 580.8 99 |         unsafe { libc::recvmmsg(sock_fd, &mut hdrs[0], count as u32, MSG_WAITFORONE, &mut ts) };
#18 580.8    |                                                                      ^^^^^^^^^^^^^^ expected `u32`, found `i32`
#18 580.8    |
#18 580.8 help: you can convert an `i32` to a `u32` and panic if the converted value doesn't fit
#18 580.8    |
#18 580.8 99 |         unsafe { libc::recvmmsg(sock_fd, &mut hdrs[0], count as u32, MSG_WAITFORONE.try_into().unwrap(), &mut ts) };
#18 580.8    |                                                                                    ++++++++++++++++++++
#18 580.8 
#18 580.9 For more information about this error, try `rustc --explain E0308`.
#18 580.9 error: could not compile `solana-streamer` due to previous error
#18 580.9 warning: build failed, waiting for other jobs to finish...
#18 601.8 error: build failed
```

MSG_WAITFORONE is defined as a `c_int`:
https://github.com/rust-lang/libc/blob/183820f6330d3d7a107632dfc87ef38212046b69/src/unix/linux_like/mod.rs#L777

--- a/src/unix/linux_like/mod.rs
+++ b/src/unix/linux_like/mod.rs
@@ -759,23 +759,95 @@ pub const PF_IEEE802154: ::c_int = AF_IEEE802154;
 pub const PF_CAIF: ::c_int = AF_CAIF;
 pub const PF_ALG: ::c_int = AF_ALG;
 
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_OOB: ::c_int = 1;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_PEEK: ::c_int = 2;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_DONTROUTE: ::c_int = 4;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_CTRUNC: ::c_int = 8;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_TRUNC: ::c_int = 0x20;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_DONTWAIT: ::c_int = 0x40;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_EOR: ::c_int = 0x80;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_WAITALL: ::c_int = 0x100;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_FIN: ::c_int = 0x200;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_SYN: ::c_int = 0x400;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_CONFIRM: ::c_int = 0x800;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_RST: ::c_int = 0x1000;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_ERRQUEUE: ::c_int = 0x2000;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_NOSIGNAL: ::c_int = 0x4000;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_MORE: ::c_int = 0x8000;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_WAITFORONE: ::c_int = 0x10000;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_FASTOPEN: ::c_int = 0x20000000;
+#[deprecated(
+    since = "0.2.136",
+    note = "recvmmsg call expects an unsigned type on musl/emscripten"
+)]
 pub const MSG_CMSG_CLOEXEC: ::c_int = 0x40000000;
 
 pub const SCM_TIMESTAMP: ::c_int = SO_TIMESTAMP;

